kubevirt_controller/tests/test_unit.py

```diff
@@ -0,0 +1,109 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import pytest
+
+from datadog_checks.dev.utils import get_metadata_metrics
+from datadog_checks.kubevirt_controller import KubeVirtControllerCheck
+
+from .conftest import mock_http_responses
+
+base_tags = [
+    "pod_name:virt-controller-some-id",
+    "kube_namespace:kubevirt",
+    "kube_cluster_name:test-cluster",
+]
+
+
+def test_emits_can_connect_one_when_service_is_up(dd_run_check, aggregator, instance, mocker):
+    mocker.patch("requests.get", wraps=mock_http_responses)
+    check = KubeVirtControllerCheck("kubevirt_controller", {}, [instance])
+    dd_run_check(check)
+    aggregator.assert_metric(
+        "kubevirt_controller.can_connect",
+        value=1,
+        tags=["endpoint:https://10.244.0.38:443/healthz", *base_tags],
+    )
+
+
+def test_emits_can_connect_zero_when_service_is_down(dd_run_check, aggregator, instance):
+    check = KubeVirtControllerCheck("kubevirt_controller", {}, [instance])
+    with pytest.raises(Exception):
+        dd_run_check(check)
+    aggregator.assert_metric(
+        "kubevirt_controller.can_connect", value=0, tags=["endpoint:https://10.244.0.38:443/healthz", *base_tags]
+    )
+
+
+def test_check_collects_all_metrics(dd_run_check, aggregator, instance, mocker):
```

Comment:
style nits (copied from my `kubevirt_api`) - could you define a list of expected metrics then iterate through the list and assert each metric has metrics_tags, and then move `base_tags` and other constants to a separate file?